src/fondant/pipeline/pipeline.py

```diff
@@ -206,7 +206,12 @@ def from_component_yaml(cls, path, **kwargs) -> "ComponentOp":
         )
 
     @classmethod
-    def from_ref(cls, ref: t.Any, **kwargs) -> "ComponentOp":
+    def from_ref(
+        cls,
+        ref: t.Any,
+        fields: t.Optional[t.Mapping[str, Field]] = None,
```

Comment:
I would prefer to keep this fields argument out of here since this is specific to the lightweight Python components. Can we move this to the `PythonComponent` class instead?

Reply:
I don't see a straightforwards way of doing this, unless we somehow pass the fields to the `BaseComponent` class since this is what the `PythonComponent` starts from but also not sure how feasible that is. Any other suggestions?

Reply:
My original implementation did this in the `Dataset` class. So I would assume we can just call a method on the `PythonComponent` class at that point?

Reply:
Ok, this might make less sense after the refactoring on main since my first commits. If we can address my comment above, it's fine for me to keep it like this for now. Would be good to add the argument to the docstring though.